packages/vite-plugin/src/proxy-middleware.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * This modules provides the proxy located at
+ * /motion-canvas-proxy/...
+ *
+ * It is needed when acessing remote resources.
+ * Trying to access remote resources works while
+ * in preview, but will fail when you try to
+ * output the image (= "read" the canvas)
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image
+ * for reasons
+ *
+ * Using the proxy circumvents CORS-issues because
+ * this way all remote resources are served from the
+ * same host as the main app.
+ */
+
+import {Connect} from 'vite';
+import axios from 'axios';
+import {ServerResponse, IncomingMessage} from 'http';
+
+/**
+ * Configuration used by the Proxy plugin
+ */
+export interface MotionCanvasCorsProxyPluginOptions {
+  /**
+   * Set which types of resources are allowed by default.
+   *
+   * @remarks
+   * Catchall on the right side is supported.
+   * Pass an empty Array to allow all types of resources, although this is not recommended.
+   *
+   * @default ["image/*", "video/*"]
+   */
+  allowedMimeTypes?: string[];
+  /**
+   * Set which hosts are allowed
+   *
+   *
+   * @remarks Note that the host is everything to the left of the first `/`, and to the right of the protocol `https://`
+   * Allowlist is not used by default, although you should consider setting up just the relevant hosts.
+   *
+   * Note that `api.iconify.design` is **always** enabled, as it is used by the Icon-Component.
+   * It gets added by the Middleware internally.
+   */
+  allowListHosts?: string[];
+}
+
+// Add Hosts required for core functionality to work
+const predefinedAllowlist = [
+  // Icons
+  'api.iconify.design',
```

Comment:
iconify is not necessary with `crossOrigin = 'anonymous'`

Reply:
The initial plan was to just reject requests going to remote addresses.

But with anon crossOrigin we can just not use the proxy if it doesn't match the allow list.

I guess the `predefinedAllowList` could be removed entirely in this case.